From 29d3b2956d5703114d324c375031c90f16942b83 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Thu, 17 May 2018 03:07:43 +0800
Subject: [PATCH 3814/5242] MGS-3892 [#imx-1027] fix dmabuf physical mapping
 issue

commit  93920e93c4d6bc289722a87aaf41e8cbbf966234 from
https://source.codeaurora.org/external/imx/linux-imx.git

g2d_buf_from_fd use gcoHAL_WrapUserMemory to wrap dmabuf handle,
and then use gcoOS_LockVideoMemory to retrieve physical address.

dmabuf is contiguous, need set contiguous flag to avoid MMU mapping,
otherwise g2d_buf_from_fd return the invalid physical address,

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../default/gc_hal_kernel_allocator_dmabuf.c       |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c
index 3883f52..d07743b 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_dmabuf.c
@@ -277,8 +277,8 @@ static int dma_buf_info_show(struct seq_file* m, void* data)
 
     Mdl->priv = buf_desc;
 
-    /* Always treat it as a non-contigous buffer. */
-    Mdl->contiguous = gcvFALSE;
+    /* Need set it as true to avoid MMU mapping. */
+    Mdl->contiguous = gcvTRUE;
 
     gcmkFOOTER_NO();
     return gcvSTATUS_OK;
-- 
1.7.9.5

